types/compat/README.md

```diff
@@ -0,0 +1,45 @@
+# Compatibility Type Aliases
+
+This directory/namespace contains compatibility type aliases for all `is_` and `validate_` functions in stdlib. They are meant as a bridge to start using Puppet 4 features and improvements, without leaving your existing code in the dust. In case there was no exact translation possible, it is noted in the comments of the respective alias.
+
+## Examples
+
+Before:
+```
+class ntp (
+  $broadcastclient,
+  $config,
+  $keys,
+  $minpoll,
+  #...
+) {
+  validate_bool($broadcastclient)
+  validate_absolute_path($config)
+  validate_array($keys)
+  if $minpoll { validate_numeric($minpoll, 16, 3) }
+}
+```
+
+
+After:
+```
+class ntp (
+  Stdlib::Compat::Bool              $broadcastclient,
+  Stdlib::Compat::Absolute_path     $config,
+  Stdlib::Compat::Array             $keys,
+  Optional[Stdlib::Compat::Numeric] $minpoll,
```

Comment:
You are right. While this is intended as a simple example, and not full-fledged code, having a `undef` default here makes sense.